### PR TITLE
FlightPlanner : Inject Custom Map tiles and reload map into custom

### PR DESCRIPTION
**Pull request by Godeffroy**
Add progress bar
Add summary of tiles injection per zoom
Load the Custom map and reload the map after injection

image image
image

### CODE REVIEW

File: GCSViews/FlightPlanner.cs [hunk 1]
```diff
@@ -1,8008 +1,8124 @@
...
-                    return;
-
-                var marker = Common.getMAVMarker(MainV2.comPort.MAV, routesoverlay);
-
-                if(marker != null)
-                    routesoverlay.Markers.Add(marker);
-
-                if (MainV2.comPort.MAV.cs.mode.ToLower() == "guided" && MainV2.comPort.MAV.GuidedMode.x != 0)
-                {
-                    addpolygonmarker(this, "Guided Mode", MainV2.comPort.MAV.GuidedMode.y / 1e
```
Comment (meee1) [inline GCSViews/FlightPlanner.cs:8051]: doesnt this just overwrite the start elements of the array populated in the previous line? it should have a destination index

**Comment by meee1** [comment]
please fix whitespace in your commit as well

**Comment by Godeffroy** [comment]
Sorry about that, i changed the code and rebase it on master.

File: GCSViews/FlightPlanner.cs [hunk 2]
```diff
@@ -8004,5 +8006,119 @@ private void gDALOpacityToolStripMenuItem_Click(object sender, EventArgs e)
...
+                    BUT_InjectCustomMap.Text = Strings.Cancel;
+                    progressBarInjectCustomMap.Value = 0;
+                    progressBarInjectCustomMap.Visible = true;
+
+                    var files_jpg = Directory.GetFiles(fbd.SelectedPath, "*.jpg", SearchOption.AllDirectories);
+                    var files_jpeg = Directory.GetFiles(fbd.SelectedPath, "*.jpeg", SearchOption.AllDirectories);
+                    var files_png = Directory.GetFiles(fbd.SelectedPath, "*.png", SearchOption.AllDirectories);
+                    string[] files = new string[files_jpg.Length + files_jpeg.Length + files_png.Length];
+                    Array.Copy(files_jpg, files, files_jpg.Length);
+                    Array.Copy(files_jpeg, files, files_jpeg.Length);
```
Comment (meee1) [inline GCSViews/FlightPlanner.cs:8049]: this is still an issue.   where is the destination index?

Comment (Godeffroy) [reply GCSViews/FlightPlanner.cs:8049]: You're absolutely right ! I'm on it.

**Pull request merged**